record-and-playback/core/lib/recordandplayback/generators/events.rb

```diff
@@ -605,6 +615,72 @@ def self.get_chat_events(events, start_time, end_time, bbb_props = {})
       chats
     end
 
+    def self.create_moderators_period(events, initial_timestamp)
+      list_moderators_period = []
+
+      events.xpath('/recording/event').each do |event|
```

Comment:
Consider using a filter on the xpath to reduce the number of events processed by the ruby code.

For example, in the generation of anonymized usernames, i do this:
```ruby
events.xpath('/recording/event[@module="PARTICIPANT" and @eventname="ParticipantJoinEvent"]').each do |event|
```

Since you want multiple different events, maybe only filter on `@module` and leave the case statement matching on the `eventname`.